### ISSUE TITLE
Lots of threads blocking in SentryHostnameCache

### ISSUE THREAD

**Issue by MaxMao** [getsentry/sentry-java#1450]
We are using version 4.1.0 in our java application. 
```
implementation 'io.sentry:sentry-spring-boot-starter:4.1.0'
```

We do not use sentry.properties file and the `sentry.dsn` system property is set to empty 
```
java -Dsentry.dsn= -jar app.jar
```

After starting up all servers were having thousands of threads and CPU usage went to 100% while serving no requests. From the thread dump we found most of the threads were `SentryHostnameCache-0`. Here is the state of one of those threads,

```
"SentryHostnameCache-0" #5256 daemon prio=5 os_prio=0 cpu=0.42ms elapsed=11676.65s tid=0x00007f717003d000 nid=0x148a waiting on condition  [0x00007f700c951000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.10/Native Method)
	- parking to wait for  <0x000000074d0f1838> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.10/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.10/AbstractQueuedSynchronizer.java:2081)
	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.10/LinkedBlockingQueue.java:433)
	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.10/ThreadPoolExecutor.java:1054)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.10/ThreadPoolExecutor.java:1114)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.10/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.10/Thread.java:834)
```

**Comment by marandaneto** [comment]
@MaxMao thanks for reporting, will check it out

cc @maciejwalkowiak

**Comment by maciejwalkowiak** [comment]
This looks like a duplicate of https://github.com/getsentry/sentry-java/issues/1263 that got fixed in https://github.com/getsentry/sentry-java/pull/1272 @MaxMao could you please upgrade to the latest SDK version?

**Comment by MaxMao** [comment]
We upgraded to the latest version last week and haven't seen the same issue since then.
